Automattic/wp-calypso
# Editor: Create a new gutenbergCustomInitialBlock state

#### Changes proposed in this Pull Request

Require: D44889-code

* Create a new `gutenbergCustomInitialBlock` modularized state that keeps track of the user's custom initial block setting, if any.
* If the user has such setting, add a `custom-initial-block` query argument to the editor iframe URL, to tell the remote site to pre-insert such block in the post content, if it's empty.

Note: there isn't any UI to set this setting yet.
You can find instructions on how to set it manually in your sandbox in D44889-code.

#### Testing instructions

* Apply D44889-code and **follow its instructions**.
* Launch Calypso on the sandboxed site, and check the Redux state.
* Make sure there's a new `gutenbergCustomInitialBlock` state, containing an item for the sandboxed site ID, with the expected custom initial block (e.g. `{ 12345678: 'core/html' }`).
* Edit a post, and check the iframe URL.
* Make sure it contains a `custom-initial-block` query arg, with the expected value (e.g. `&custom-initial-block=core/html`).

Fixes #
